alpaka/include/alpaka/math/max/MaxUniformCudaHipBuiltIn.hpp

```diff
@@ -71,6 +71,9 @@ namespace alpaka
                     MaxUniformCudaHipBuiltIn const & max_ctx,
                     Tx const & x,
                     Ty const & y)
+#ifdef _MSC_VER
+                    -> Tx // FIXME(bgruber): return type is deduced as void by MSVC as host compiler (nvcc deduces correct return type)
+#endif
```

Comment:
I have no clue how to solve this. Maybe it is an issue with MSVC. Maybe also in the CUDA SDK for windows. Any ideas?

Reply:
Funnily enough, I just tried the whole thing on Gentoo with CUDA 10.2 and have the same issue with g++8.4 as host compiler:
`error: void value not ignored as it ought to be`

Reply:
I do not have an idea, but can try on my local machine with MSVS + CUDA. It may be that something does wrong with `::max(x, y)` that MSVS and g++ 8.4 for some reason think it returns `void`?

Reply:
I eventually just explicitely specified the return type as `decltype(::max(x, y))`. It seems this fixes the misscompilation.

Reply:
Is this change only a preview?
Please do not change any line in a subtree, this change will get lost after with the next subtree update.

Reply:
I changed this in the alpaka subtree to fix compilation for MSVC and g++. I think it needs to be addressed within alpaka. Here is the issue: https://github.com/alpaka-group/alpaka/issues/1013